[quant] Fix qmin/qmax when using customized qrange

terrychenism opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #74717

Summary:
currently the weight map to 0 and max_float to 65535 due to incorrect qmin/qmax in qin16 customized qrange
the expectation from the set observers is the integer representation is supposed to be a signed int16 i.e -32768 to 32767.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35129924](https://our.internmc.facebook.com/intern/diff/D35129924)

terrychenism committed:
[quant] Fix qmin/qmax when using customized qrange
Summary:
currently the weight map to 0 and max_float to 65535 due to incorrect qmin/qmax in qin16 customized qrange
the expectation from the set observers is the integer representation is supposed to be a signed int16 i.e -32768 to 32767.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

terrychenism commented:
@terrychenism has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35129924).

terrychenism committed:
Update on "[quant] Fix qmin/qmax when using customized qrange"


Summary:
currently the weight map to 0 and max_float to 65535 due to incorrect qmin/qmax in qin16 customized qrange
the expectation from the set observers is the integer representation is supposed to be a signed int16 i.e -32768 to 32767.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35129924](https://our.internmc.facebook.com/intern/diff/D35129924)

[ghstack-poisoned]

terrychenism commented:
@terrychenism has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35129924).

jerryzh168 reviewed (approved):
thanks! please make sure ci passes

terrychenism committed:
Update on "[quant] Fix qmin/qmax when using customized qrange"


Summary:
currently the weight map to 0 and max_float to 65535 due to incorrect qmin/qmax in qin16 customized qrange
the expectation from the set observers is the integer representation is supposed to be a signed int16 i.e -32768 to 32767.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35129924](https://our.internmc.facebook.com/intern/diff/D35129924)

[ghstack-poisoned]

terrychenism commented:
@terrychenism has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35129924).

jerryzh168 commented on torch/ao/quantization/utils.py:
```diff
@@ -326,10 +326,6 @@ def calculate_qmin_qmax(quant_min: int, quant_max: int, has_customized_qrange: b
-        if dtype == torch.qint8:
-            quant_min, quant_max = -qrange_len // 2, qrange_len // 2 - 1
-        else:
-            quant_min, quant_max = 0, qrange_len - 1
```
I think you can remove the next line as well

terrychenism replied:
do you mean remove 'reduce_range' if statement?

jerryzh168 replied:
yes, L333-L334

terrychenism replied:
if we remove it, how to handle the qmin/max if user wanna use has_customized_qrange with reduce_range flag?

terrychenism replied:
it would break several test cases if we remove it.

jerryzh168 replied:
user should set the range correctly in the beginning directly I think?

terrychenism committed:
Update on "[quant] Fix qmin/qmax when using customized qrange"


Summary:
currently the weight map to 0 and max_float to 65535 due to incorrect qmin/qmax in qin16 customized qrange
the expectation from the set observers is the integer representation is supposed to be a signed int16 i.e -32768 to 32767.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35129924](https://our.internmc.facebook.com/intern/diff/D35129924)

[ghstack-poisoned]

terrychenism committed:
Update on "[quant] Fix qmin/qmax when using customized qrange"


Summary:
currently the weight map to 0 and max_float to 65535 due to incorrect qmin/qmax in qin16 customized qrange
the expectation from the set observers is the integer representation is supposed to be a signed int16 i.e -32768 to 32767.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35129924](https://our.internmc.facebook.com/intern/diff/D35129924)

[ghstack-poisoned]

terrychenism committed:
Update on "[quant] Fix qmin/qmax when using customized qrange"


Summary:
currently the weight map to 0 and max_float to 65535 due to incorrect qmin/qmax in qin16 customized qrange
the expectation from the set observers is the integer representation is supposed to be a signed int16 i.e -32768 to 32767.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35129924](https://our.internmc.facebook.com/intern/diff/D35129924)

[ghstack-poisoned]

terrychenism committed:
Update on "[quant] Fix qmin/qmax when using customized qrange"


Summary:
currently the weight map to 0 and max_float to 65535 due to incorrect qmin/qmax in qin16 customized qrange
the expectation from the set observers is the integer representation is supposed to be a signed int16 i.e -32768 to 32767.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35129924](https://our.internmc.facebook.com/intern/diff/D35129924)

[ghstack-poisoned]

terrychenism commented:
@terrychenism has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35129924).

terrychenism committed:
Update on "[quant] Fix qmin/qmax when using customized qrange"


Summary:
currently the weight map to 0 and max_float to 65535 due to incorrect qmin/qmax in qin16 customized qrange
the expectation from the set observers is the integer representation is supposed to be a signed int16 i.e -32768 to 32767.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35129924](https://our.internmc.facebook.com/intern/diff/D35129924)

[ghstack-poisoned]

terrychenism commented:
@terrychenism has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35129924).

terrychenism committed:
Update on "[quant] Fix qmin/qmax when using customized qrange"


Summary:
currently the weight map to 0 and max_float to 65535 due to incorrect qmin/qmax in qin16 customized qrange
the expectation from the set observers is the integer representation is supposed to be a signed int16 i.e -32768 to 32767.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35129924](https://our.internmc.facebook.com/intern/diff/D35129924)

[ghstack-poisoned]

terrychenism commented:
@terrychenism has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35129924).

terrychenism committed:
Update on "[quant] Fix qmin/qmax when using customized qrange"


Summary:
currently the weight map to 0 and max_float to 65535 due to incorrect qmin/qmax in qin16 customized qrange
the expectation from the set observers is the integer representation is supposed to be a signed int16 i.e -32768 to 32767.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35129924](https://our.internmc.facebook.com/intern/diff/D35129924)

[ghstack-poisoned]

terrychenism commented:
@terrychenism has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35129924).
